ui/page/dcrdex/dex_onboarding_page.go

```diff
@@ -844,76 +892,207 @@ func (pg *DEXOnboarding) HandleUserInteractions() {
-				return !a.IsWatchOnly
-			})
-			pg.bondSourceAccountSelector.HideLogo = true
-			if err := pg.bondSourceAccountSelector.SelectFirstValidAccount(pg.bondSourceWalletSelector.SelectedWallet()); err != nil {
-				log.Error(err)
+				pg.bondSourceAccountSelector.HideLogo = true
+				if err := pg.bondSourceAccountSelector.SelectFirstValidAccount(pg.bondSourceWalletSelector.SelectedWallet()); err != nil {
+					log.Error(err)
+				}
+
+				pg.bondStrengthEditor.Editor.SetText(fmt.Sprintf("%d", minimumBondStrength))
+				pg.newTier = minimumBondStrength
+			}
+
+			// Proceed to next step if we already have the dex pass cached.
+			if len(pg.dexPass) > 0 {
+				go nextStep()
+				break
+			}
+
+			if !pg.core.IsDEXPasswordSet() {
+				// Fresh onboarding process.
+				pg.isLoading = true
+				go func() {
+					defer func() {
+						pg.isLoading = false
+					}()
+
+					pg.dexPass = []byte(pg.passwordEditor.Editor.Text())
+					if err := pg.core.SetDEXPassword(pg.dexPass, nil); err != nil {
+						pg.notifyError(err.Error())
+						return
+					}
+
+					nextStep()
+					pg.ParentWindow().Reload()
+				}()
+				break
 			}
 
-			pg.bondStrengthEditor.Editor.SetText(fmt.Sprintf("%d", minimumBondStrength))
-			pg.newTier = minimumBondStrength
+			// User already has dex password set but did not finish the
+			// onboarding.
+			dexPasswordModal := modal.NewCreatePasswordModal(pg.Load).
+				EnableName(false).
+				EnableConfirmPassword(false).
+				Title(values.String(values.StrDexPassword)).
+				SetPositiveButtonCallback(func(_, password string, pm *modal.CreatePasswordModal) bool {
+					pg.dexPass = []byte(password)
+					err := pg.core.Login(pg.dexPass)
+					if err != nil {
+						pm.SetError(err.Error())
+						pm.SetLoading(false)
+						return false
+					}
 
+					nextStep()
+					pm.Dismiss()
+					pg.ParentWindow().Reload()
+					return true
+				})
+			pg.ParentWindow().ShowModal(dexPasswordModal)
 		case onboardingPostBond:
+			if pg.isLoading {
+				break
+			}
+
 			// Validate all input fields.
 			hasEnough := pg.bondAccountHasEnough()
 			bondStrengthOk := pg.validateBondStrength()
 			if !hasEnough || !bondStrengthOk {
-				return
+				break
+			}
+
+			asset := pg.bondSourceWalletSelector.SelectedWallet()
+			bondAsset := pg.bondServer.bondAssets[asset.GetAssetType()]
+			postBond := &core.PostBondForm{
+				Addr:      pg.bondServer.url,
+				AppPass:   pg.dexPass,
+				Asset:     &bondAsset.ID,
+				Bond:      uint64(pg.newTier) * bondAsset.Amt,
+				Cert:      pg.bondServer.cert,
+				FeeBuffer: pg.core.BondsFeeBuffer(bondAsset.ID),
 			}
 
-			// TODO: Post bond, wait for confirmations and redirect to market page.
-			pg.currentStep = onBoardingStepWaitForConfirmation
-			// Scroll to the top of the confirmation page after leaving the long
-			// post bond form.
-			pg.scrollContainer.Position.Offset = 0
+			pg.isLoading = true
+			pg.bondConfirmationInfo = new(bondConfirmationInfo)
```

Comment:
This should probably be nil until after `core.PostBond` completes and returns a response.